### PR TITLE
Civ 4284 your details an other type

### PR DESCRIPTION
**Pull request by laarmada**
### JIRA link ###

https://tools.hmcts.net/jira/browse/CIV-4284

### Change description ###

Enter organisation details' screen

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/routes/features/claim/yourDetails/claimantDoBController.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    } else {
+      const claim = await getCaseDataFromStore(req.params.id);
+      if (claim.respondent1) {
+        claim.respondent1.dateOfBirth = form.model.dateOfBirth;
+      } else {
+        const respondent = new Party();
+        respondent.dateOfBirth = form.model.dateOfBirth;
+        claim.respondent1 = respondent;
+      }
+      await saveDraftClaim(req.params.id, claim);
```
Comment (laperla11) [inline src/main/routes/features/claim/yourDetails/claimantDoBController.ts:45]: req.params.id is not available for claim journey, you need req.session.user.id here

**Review by laperla11 — CHANGES_REQUESTED** [review]

File: src/main/routes/features/claim/yourDetails/claimantDoBController.ts (hunk 1 above)
Comment (portizdev) [reply src/main/routes/features/claim/yourDetails/claimantDoBController.ts]: Done!

**Review by portizdev — APPROVED** [review]

**Review by laperla11 — APPROVED** [review]

File: src/test/unit/views/features/claim/yourDetails/claimant-organisation-details.test.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+const pageTitle = 'PAGES.CLAIM_JOURNEY.CLAIMANT_INDIVIDUAL_DETAILS.PAGE_TITLE';
+
+jest.mock('../../../../../../main/modules/oidc');
+jest.mock('../../../../../../main/modules/draft-store');
+
+describe('Claimant Organisation Details View', () => {
+  const citizenRoleToken: string = config.get('citizenRoleToken');
+  const idamUrl: string = config.get('idamUrl');
+
+  beforeEach(() => {
```
Comment (commitBlob) [inline src/test/unit/views/features/claim/yourDetails/claimant-organisation-details.test.ts:20]: can you change this to `beforeAll` please. We don't need to call this for every test

File: src/test/unit/views/features/claim/yourDetails/claimant-organisation-details.test.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      .post('/o/token')
+      .reply(200, {id_token: citizenRoleToken});
+    app.locals.draftStoreClient = mockCivilClaim;
+  });
+
+  describe('on GET', () => {
+    let htmlDocument: Document;
+    let mainWrapper: Element;
+
+    beforeEach(async () => {
```
Comment (commitBlob) [inline src/test/unit/views/features/claim/yourDetails/claimant-organisation-details.test.ts:31]: can you change this to `beforeAll` please. We don't need to call this for every test as page content doesn't change

**Review by TatsyMaslova — APPROVED** [review]

File: src/main/routes/features/claim/yourDetails/defendantPartyTypeController.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import {CLAIM_DEFENDANT_PARTY_TYPE_URL} from '../../../urls';
+import {GenericForm} from '../../../../common/form/models/genericForm';
+import {PartyTypeSelection} from '../../../../common/form/models/claim/partyTypeSelection';
+import {redirectToPage} from '../../../../services/features/claim/partyTypeService';
+import {ClaimantOrDefendant} from '../../../../common/models/partyType';
+import {
+  getDefendantInformation,
+  saveDefendant,
+} from '../../../../services/features/claim/yourDetails/defendantDetailsService';
+import {Party} from 'models/party';
```
[the remark this answers is not in the document]
Comment (portizdev) [reply src/main/routes/features/claim/yourDetails/defendantPartyTypeController.ts]: Sometimes it fails me, in this case it was working but for consistency I have changed it to the full route.

File: src/test/unit/views/features/claim/yourDetails/claimant-organisation-details.test.ts (hunk 3 above)
Comment (portizdev) [reply src/test/unit/views/features/claim/yourDetails/claimant-organisation-details.test.ts]: Done

**Pull request merged**